src/main/java/com/softteam/vocabuilder/controller/VocabularyController.java

```diff
@@ -18,36 +21,75 @@ public class VocabularyController {
 
     @Autowired
     private VocabularyServiceImp vocabularyServiceImp;
-    @Autowired
-    private VocabularyRepository vocabularyRepository;
 
     @PostMapping
-    public ResponseEntity<Vocabulary> createVocabulary(@RequestBody Vocabulary vocabulary){
-        vocabularyServiceImp.create(vocabulary);
+    public ResponseEntity<Vocabulary> createVocabulary(@RequestBody VocabularyDTO vocabularyDTO) {
+        //Internal Server Error al faltar un columna
+        if (vocabularyDTO.getTitle() == null || vocabularyDTO.getDescription() == null) {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        vocabularyServiceImp.create(vocabularyDTO);
         return new ResponseEntity<Vocabulary>(HttpStatus.CREATED);
     }
 
     @PutMapping("/{id}")
-    public ResponseEntity<Vocabulary> updateVocabulary(@PathVariable(value = "id")UUID id, @RequestBody Vocabulary vocabulary){
-        vocabulary.setId(id);
-        vocabularyServiceImp.update2(vocabulary);
+    public ResponseEntity<Vocabulary> updateVocabulary(@PathVariable(value = "id") String id, @RequestBody Vocabulary vocabulary) {
+        UUID uuidID = UuidUtil.validateUUIDType(id);
+        Optional<Vocabulary> vocabulary1 = Optional.of(new Vocabulary());
+        try {
+            vocabulary1 = vocabularyServiceImp.getVocabulary(uuidID);
+        } catch (RuntimeException e) {
+            throw new VocabularyNotFoundException(e.getMessage(), HttpStatus.NO_CONTENT);
+        }
+        if (vocabulary1.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        }
+        if (vocabulary.getTitle() == null) {
+            vocabulary.setTitle(vocabulary1.get().getTitle());
+        }
+        if (vocabulary.getDescription() == null) {
+            vocabulary.setDescription(vocabulary1.get().getDescription());
+        }
+        vocabulary.setId(uuidID);
```

Comment:
esto estaria bien en el `Patch` method, que es un update partial, pero en este lo que manden es lo que hay que actualizar, no si es diferente de null, si lo que no quieres es que no te manden null, seria una validacion como la del metodo create

Reply:
aqui estaba garantizando saber que era lo que estaba modificando el usuario o el titulo o la descripcion o los dos o ninguno.
Entonces otra duda:
```
        Optional<Vocabulary> vocabulary1 = Optional.of(new Vocabulary());
        try {
            vocabulary1 = vocabularyServiceImp.getVocabulary(uuidID);
        } catch (RuntimeException e) {
            throw new VocabularyNotFoundException(e.getMessage(), HttpStatus.NO_CONTENT);
        }
```
aqui en esa parte del codigo llamo a `vocabularyServiceImp.getVocabulary(uuidID)` esta llamada para saber si el objeto se encuentra ya guardado y lo estoy usando en `updateVocabulary` , `getVocabulary` y `deleteVocabuilder` que este ultimo de paso esta mal puesto el nombre...entoces el `vocabularyServiceImp.getVocabulary(uuidID)` es el unico que en el Service que laza una execcion 
```    public Optional<Vocabulary> getVocabulary(UUID id) {
        Optional<Vocabulary> optionalVocabulary = vocabularyRepository.findById(id);
        if (optionalVocabulary.isEmpty()) {
            throw new RuntimeException("vocabulary not found");
        }
        return optionalVocabulary;
    }
```
no se hasta que punto esta bien todo eso

Reply:
Entoces la `RuntimeException("vocabulary not found")` del metodo anterior la capturo en el primer fracmento de codigo del metodo `updateVocabulary` en mi controller